android/src/main/java/com/alternadom/wifiiot/WifiIotPlugin.java

```diff
@@ -173,6 +186,50 @@ public void onDetachedFromActivity() {
         moActivity = null;
     }
 
+    @Override
+    public boolean onRequestPermissionsResult(int requestCode, String[] permissions,
+                                              int[] grantResults) {
+        switch (requestCode) {
```

Comment:
Should compute whether the permission was granted before the `switch` statement.

```java
final bool wasPermissionGranted = grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED;
```

Reply:
Done, though it will have to be changed if we ever request multiple permissions at once.

Reply:
Understood. Even then, it'll still be easier since you just have to create a variation of that line based on the permission you want to verify.

Even better, you can also create a function that checks the value at a given index, passing in the grant results and the index you want to interrogate.